src/prometheus.ml

```diff
@@ -81,11 +79,11 @@ module MetricFamilyMap = Map.Make(MetricInfo)
 
 module CollectorRegistry = struct
   type t = {
-    mutable metrics : (unit -> (string * float) list LabelSetMap.t) MetricFamilyMap.t;
+    mutable metrics : (unit -> (string * float * ((LabelName.t * float) option)) list LabelSetMap.t) MetricFamilyMap.t;
     mutable pre_collect : (unit -> unit) list;
   }
 
-  type snapshot = (string * float) list LabelSetMap.t MetricFamilyMap.t
+  type snapshot = (string * float * ((LabelName.t * float) option)) list LabelSetMap.t MetricFamilyMap.t
```

Comment:
Why do we need a third component to the tuple here? It looks like we have a list, plus an optional extra element to add to the list, which seems pointless.

Reply:
The string bit of the list contains a suffix for the metric name, the added parameter is for a label.
i.e.:
["", 3.0, Some ("blah", 3.0)] ->  "metricname{blah="3.0"} 3.0"
["_sum", 3.0, None] -> "metricname_sum 3.0"